### PR TITLE
Setup flipper using warden-github pattern

### PR DESCRIPTION
**Pull request by ryan-mcneil**
## Summary

- This PR changes the way we provide access to the Flipper UI, from some complicated methods using a hardcoded list of users (`Settings.flipper.admin_user_emails`) to GitHub Team-based OAuth, similar to how we've done it across a few other areas of the app (ArgoCD/Terminal access, Sidekiq UI, Coverband, etc).
- The user will now need to be a member of ~~the respective GitHub teams for each environment (below)~~ the `VA.gov-flipper-prod` team to perform any actions in the Flipper UI (Otherwise, they can only view status of features, but not toggle them or see info on the "show" page.
- Upon navigating to a view (`/flipper/features` or `/flipper/features/:my_feature`, the user will be prompted to Sign in to GitHub. If the user signs in, but aren't a part of the `department-of-veterans-affairs` org or correct GitHub Team, they get another error message, linking them to some supporting docs. If authorized, they'll have normal access to the UI

GitHub Teams:
* ~~[VA.gov-flipper-dev](https://github.com/orgs/department-of-veterans-affairs/teams/va-gov-flipper-dev)~~
* ~~[VA.gov-flipper-sandbox](https://github.com/orgs/department-of-veterans-affairs/teams/va-gov-flipper-sandbox)~~
* [VA.gov-flipper-staging](https://github.com/orgs/department-of-veterans-affairs/teams/va-gov-flipper-staging)
* [VA.gov-flipper-prod](https://github.com/orgs/department-of-veterans-affairs/teams/va-gov-flipper-prod)
NOTE: At this time we only intend on using two teams: `VA.gov-flipper-prod` for Prod, and `VA.gov-flipper-staging` for all lesser environments

Things to do before merging:
- [x] Add users to each of the respective GitHub teams
- [x] Create new access request template (VA.gov-flipper-prod)
- [x] Update Documentation
- [ ] Add values to Devops repo

Things to test in Staging:
- [ ] Authentication flow works properly
- [ ] GitHub logo asset appears as desired

## Related issue(s)
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/54110

## Testing done

- Tested locally, using a GitHub Oauth App and GitHub Team, both belonging to the `department-of-veterans-affairs` org.
- Tested signing in (unauthenticated, authenticated but missing team membership, fully authorized), navigating, and toggling, plus removing browser session, confirming that it continued to secure endpoints and errors were handled properly

## Screenshots
https://github.com/department-of-veterans-affairs/vets-api/assets/41602677/f95e0082-5fe5-42f3-a683-654829e19c78

## What areas of the site does it impact?
* Flipper UI (/flipper/features)

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [x]  I added a screenshot of the developed feature

### CODE REVIEW

**Comment by jhouse-solvd** [comment]
Please don't let my observations or suggestions stand in the way of shipping this, but thought I'd share: 

Overall, I love the aim of this PR and the problem it's solving. It seems like a good step toward simplifying access to Platform tools. 👍 

From a security perspective, the Platform has been moving toward intentionally limiting access to production across various pieces of the developer ecosystem. I don’t know if that applies to Feature Flipper, but it seems like worthwhile design consideration for this or a future iteration. 

If there is a desire to limit access to production Feature Flipper _and_ simplify the total number of groups to administer, another approach could be to have two (2) groups: prod & non-prod (ie, lowers). This might still provide good separation and make it easier to audit _(assuming there's a supporting process to request and request, review, and grant access to Feature Flipper in production)_. 
 
cc: @little-oddball @kenmayo

File: lib/flipper/views/features.erb [hunk 1]
```diff
@@ -20,7 +27,8 @@
               <%= feature.state.capitalize %>
             </td>
             <td>
-              <% if RequestStore.store[:flipper_user_email_for_log].present? || Rails.env.development? %>
+              <%# TODO Add `|| Rails.env.development?` back to line below %>
```
Comment (ericboehs) [inline lib/flipper/views/features.erb:30]: Delete me

**Review by ericboehs — DISMISSED** [review]
Might want to add a way to sign out.

Looks great though! Thanks for walking me through this.

File: config/settings.yml [hunk 2]
```diff
@@ -884,196 +884,6 @@ search_click_tracking:
   mock: false
   url: https://api.gsa.gov/technology/searchgov/v2
 
-flipper:
-  mute_logs: false
-  admin_user_emails:
```
Comment (rmtolmach) [inline config/settings.yml:889]: 👏 so satisfying!!!

**Review by rmtolmach — APPROVED** [review]

**Pull request merged**